conductr_cli/bndl_main.py

```diff
@@ -0,0 +1,597 @@
+
+                    with gzip.GzipFile(fileobj=dest_file_digest, mode='wb', compresslevel=0) as dest:
+                        shutil.copyfileobj(fileobj, dest)
+
+                    dest_file_hexdigest = dest_file_digest.digest_out.hexdigest()
+                    dest_file_size = dest_file_digest.size_out
+                    digests[dest_file_hexdigest] = dest_path
+                    sizes[dest_file_hexdigest] = dest_file_size
+                    layers_to_digests[os.path.join(immediate_parent_dir, file_name)] = dest_file_hexdigest
+
+                os.renames(dest_path, '{}/blobs/sha256/{}'.format(destination, dest_file_hexdigest))
+            else:
+                with open(os.path.join(temp_dir, 'layers', parent_dir, file_name), 'wb') as dest:
+                    shutil.copyfileobj(fileobj, dest)
+
+        if is_dir:
+            for base, dirs, files in os.walk(data):
+                for file in files:
+                    name = os.path.join(base, file)
+
+                    with open(name, 'rb') as fileobj:
+                        handle_entry(name, fileobj, isdir=False, isfile=True)
+        else:
+            for entry in data:
+                if entry.isfile():
+                    handle_entry(entry.name, data.extractfile(entry), isdir=False, isfile=True)
+
+        contents_dir = None
+
+        for base, dirs, files in os.walk(temp_dir):
+            if 'manifest.json' in files:
+                contents_dir = base
+                break
+
+        if contents_dir is None:
+            raise RuntimeError('Missing manifest.json for docker format')
+
+        with open(os.path.join(contents_dir, 'manifest.json'), 'r') as manifest_file:
+            manifests = json.load(manifest_file)
+            manifest = None
+
+            for m in manifests:
+                if m['RepoTags'] and any(t for t in m['RepoTags'] if docker_image_name_matches(args.name, args.tag, t)):
+                    manifest = m
+                    break
+
+            image_name, image_tag = docker_parse_image_name(manifest['RepoTags'][0])
+
+            with open(os.path.join(contents_dir, manifest['Config'])) as config_file:
+                config = json.load(config_file)
+
+                oci_spec = docker_config_to_oci_image(manifest, config, sizes, layers_to_digests)
+
+                file_write_bytes(
+                    '{}/blobs/sha256/{}'.format(destination, oci_spec['config_digest']),
+                    oci_spec['config']
+                )
+
+                file_write_bytes(
+                    '{}/blobs/sha256/{}'.format(destination, oci_spec['manifest_digest']),
+                    oci_spec['manifest']
+                )
+
+                file_write_bytes(
+                    '{}/refs/{}'.format(destination, image_tag),
+                    oci_spec['refs']
+                )
+
+            return image_name
+    finally:
+        shutil.rmtree(temp_dir)
```

Comment:
I agree.

My suggestion would be:

*Move the `bndl` method into its own `bndl.py` and rename it to `create_bundle` or something similar.*

This way it can be called from `bndl_main.py` as `bndl.create_bundle`. This will be consistent with the `conduct` and `sandbox` command where the function that's doing the actual work is encapsulated into its own module (i.e. `conduct_info` module for `conduct info` command).

*Move the Docker specific functionality into `bndl_docker` and OCI into `bndl_oci`**

Then these can be called from `bndl.create_bundle`.